tags_exec.go

```diff
@@ -0,0 +1,77 @@
+package pongo2
+
+import (
+	"bytes"
+	"log"
+)
+
+type tagExecNode struct {
+	position    *Token
+	bodyWrapper *NodeWrapper
+}
+
+func (node *tagExecNode) Execute(ctx *ExecutionContext, writer TemplateWriter) *Error {
+	temp := bytes.NewBuffer(make([]byte, 0, 1024)) // 1 KiB size
+
+	err := node.bodyWrapper.Execute(ctx, temp)
+	if err != nil {
+		return err
+	}
+	templateSet := ctx.template.set
+	s := temp.String()
+	currentTemplate, _ := templateSet.FromString(s)
+	if currentTemplate == nil {
+		log.Fatalf("Could not read template %s, please check the syntax", s)
+	}
+
+	// New context with global, public and private context
+	// we need all 3, globals comes from templateSet,
+	// public for all variables passed, private for macros etc defined in amin template
+	newContext := make(Context)
+	newContext.Update(ctx.Private)
+	newContext.Update(ctx.Public)
+	finalRes, _ := currentTemplate.Execute(newContext)
+	moveMacrosToMainTemplate(currentTemplate, ctx)
+
+	_, err2 := writer.WriteString(finalRes)
+	if err2 != nil {
+		return nil
```

Comment:
```suggestion
        currentTemplate.nodeDocument.Execute(ctx, writer)
```
Have we tried the above?

Reply:
Tried. it is working as expected.